draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -810,32 +968,38 @@ Procedure:
 15. return (A, e, s)
 ```
 
-### SignatureToOctets
+### ProofToOctets
 
-This operation describes how to encode a signature to an octet string.
+This operation describes how to encode a proof, as computed at step 25 in [ProofGen](#proofgen), to an octet string. The input to the operation MUST be a valid proof.
 
-*Note* this operation deliberately does not perform the relevant checks on the inputs `A`, `e` and `s`
-because its assumed these are done prior to its invocation, e.g as is the case with the [Sign](#sign) operation.
+The inputted proof value must consist of the following components, in that order:
+
+1. Three (3) valid points of the G1 subgroup, different from the identity point of G1 (i.e., `A', Abar, D`, in ProofGen)
+2. Five (5) integers representing scalars in the range of 1 to r-1 inclusive (i.e., `c, e^, r2^, r3^, s^`, in ProofGen).
+3. A number of integers representing scalars in the range of 1 to r-1 inclusive, corresponding to the undisclosed from the proof messages (i.e., `m^_j1, ..., m^_jU`, in ProofGen, where U the number of undisclosed messages).
 
 ```
-signature_octets = signature_to_octets(signature)
+proof_octets = proof_to_octets(proof)
 
 Inputs:
 
-- signature (REQUIRED), a valid signature, in the form (A, e, s), where
-                        A a point in G1 and e, s non-zero scalars mod r.
+- proof (REQUIRED), a BBS proof in the form calculated by ProofGen in
+                    step 27 (see above).
+
+Parameters:
+
+- octet_scalar_length (REQUIRED), non-negative integer. The length of
+                                  a scalar octet representation, defined
+                                  by the ciphersuite.
 
 Outputs:
 
-- signature_octets, octet string.
+- proof_octets, octet string or INVALID.
 
 Procedure:
 
-1. (A, e, s) = signature
-2. A_octets = point_to_octets_g1(A)
-3. e_octets = I2OSP(e, octet_scalar_length)
-4. s_octets = I2OSP(s, octet_scalar_length)
-5. return (A_octets || e_octets || s_octets)
+1. (A', Abar, D, c, e^, r2^, r3^, s^, (m^_1, ..., m^_U)) = proof
```

Comment:
Not a bad idea to add some redundancy in the proof value. Would also make the suggestion in #227 easier to implement.

That said, since `ProofVerify` is not a terribly inefficient operation, not sure how much will provide beyond some useful error information? Also, is truncating the proof something that will (somewhat) "commonly" happen (i assume not as much in the application layer)??

In any case it seems that it would be better addressed in a separate PR, so will open an issue to track the conversation and will resolve this comment if that is ok? Please feel free to un-resolve!

Reply:
opened #233 to track the issue!